chain/network/src/time.rs

```diff
@@ -89,63 +90,102 @@ impl Clock {
     pub async fn sleep_until(&self, t: Instant) {
         match &self.0 {
             ClockInner::Real => tokio::time::sleep_until(t.into_inner().into()).await,
-            ClockInner::Fake(fake) => fake.advance_until(t),
+            ClockInner::Fake(fake) => fake.sleep_until(t).await,
         }
     }
 
     /// Cancellable.
     pub async fn sleep(&self, d: Duration) {
         match &self.0 {
             ClockInner::Real => tokio::time::sleep(d.try_into().unwrap()).await,
-            ClockInner::Fake(fake) => fake.advance(d),
+            ClockInner::Fake(fake) => fake.sleep(d).await,
         }
     }
 }
 
 struct FakeClockInner {
-    mono: Instant,
+    mono: watch::Sender<Instant>,
```

Comment:
can you add a comment on why we need to use watch::Sender ?

Reply:
done